Applications/Spire/Source/Ui/TextBox.cpp

```diff
@@ -184,7 +183,9 @@ connection
 }
 
 QSize TextBox::sizeHint() const {
-  return scale(160, 30);
+  auto size_hint = QSize(m_line_edit->fontMetrics().horizontalAdvance(
+      m_model->get_current()) + 5, m_line_edit->fontMetrics().height());
```

Comment:
Why add 5 to the width?

Reply:
Qt adds a fixed horizontalMargin of 2 to both sides of the text, and adds another pixel to the right side of the text for the cursor. These 5 extra pixels are always there regardless of the padding of the QLineEdit, so I add it back so that when the TextBox's size hint is calculated it doesn't result in the text 'scrolling' behind these extra margins. Qt doesn't scale these values so I simply add '+ 5' to the size hint's width.

It would be preferable to be able to remove these margins but aside from implementing a custom QStyle or painting the line edit, it doesn't appear like there's a simple solution so it was decided to maintain these margins.